DemoApp/StreamChat/Components/CustomAttachments/LocationAttachment/LocationAttachmentSnapshotView.swift

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import MapKit
+import StreamChatUI
+import UIKit
+
+class LocationAttachmentSnapshotView: _View {
+    static var snapshotsCache: NSCache<NSString, UIImage> = .init()
+
+    var coordinate: LocationCoordinate? {
+        didSet {
+            updateContent()
+        }
+    }
+
+    var snapshotter: MKMapSnapshotter?
+
+    var didTapOnLocation: (() -> Void)?
+
+    lazy var imageView: UIImageView = {
+        let view = UIImageView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.isUserInteractionEnabled = true
+        view.clipsToBounds = true
+        view.layer.cornerRadius = 16
+        view.contentMode = .scaleAspectFill
+        return view
+    }()
+
+    lazy var activityIndicatorView: UIActivityIndicatorView = {
+        let view = UIActivityIndicatorView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.style = .medium
+        return view
+    }()
+
+    let mapOptions: MKMapSnapshotter.Options = .init()
+
+    override func setUp() {
+        super.setUp()
+
+        let tapGestureRecognizer = UITapGestureRecognizer(
+            target: self,
+            action: #selector(handleTapOnWorkoutAttachment)
+        )
+        imageView.addGestureRecognizer(tapGestureRecognizer)
+    }
+
+    override func setUpLayout() {
+        super.setUpLayout()
+
+        addSubview(activityIndicatorView)
+        addSubview(imageView)
+
+        NSLayoutConstraint.activate([
+            imageView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            imageView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            imageView.topAnchor.constraint(equalTo: topAnchor),
+            imageView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            activityIndicatorView.centerXAnchor.constraint(equalTo: imageView.centerXAnchor),
+            activityIndicatorView.centerYAnchor.constraint(equalTo: imageView.centerYAnchor)
+        ])
+    }
+
+    @objc func handleTapOnWorkoutAttachment() {
+        didTapOnLocation?()
+    }
+
+    override func updateContent() {
```

Comment:
this method is a bit long and hard to follow. Would be nice to split it in smaller methods, like setting region, calculating the point of the pin etc.

Reply:
Yeah, I did not focus much on it because it is Demo code, but I'll improve the readability

Reply:
Done ✅

Reply:
much better 👍